hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
Sure, makes sense then. You could also just wrap it on an `Instance`, so that consumers would call `#get()` and CDI resolution would happen then. In any case, I'm OK with this code and this can be refactored later if problems arise. Perhaps then just add a comment on the code pointing to this discussion here?